url7606378bcc_ljy2010a_protobuf2_tgz-pJ8-com_game_gameObj_bufBuilder_ReflectBuilderJ8/src/com/game/gameObj/bufBuilder/BufBuilderMake.java

```diff
@@ -0,0 +1,140 @@
+package com.game.gameObj.bufBuilder;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class BufBuilderMake {
+
+	public static String[] typeNames = {"Soldier","WarShip"};
+	
+	/**
+	 * @param args
+	 * @throws ClassNotFoundException 
+	 * @throws IOException 
+	 */
+	public static String FilePath = "src/com/game/gameObj/bufBuilder/";
+	
+	public static String GameObjId  = "GameObjId";
+	
+	public static String GameObjType  = "GameObjType";
+	
+	public static String Coor = "Coor";
+	
+	
+	public static void main(String[] args) throws ClassNotFoundException, IOException {
+		// TODO Auto-generated method stub
+		for (int i = 0 ;i<typeNames.length;i++){
+			PrintWriter pw = new PrintWriter( new FileWriter( FilePath+ typeNames[i] + "Builder.java" ) );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *pw*. The resource is referenced at lines 28, 29. The resource is closed at line 30. There are other execution paths that do not contain closure statements, for example, when BufBuilderMake.makeBuilder throws an exception. Ensure *pw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).